src/gf.c

```diff
@@ -1039,9 +1056,35 @@ jl_lambda_info_t *jl_get_specialization1(jl_tupletype_t *types)
     return NULL;
 }
 
-JL_DLLEXPORT void jl_compile_hint(jl_tupletype_t *types)
+// Specialize, but return NULL if there is a call ambiguity
+jl_lambda_info_t *jl_checked_specialization1(jl_tupletype_t *types)
```

Comment:
I'm struggling with this a bit. `codegen.cpp` sometimes gets `NULL` back from `jl_get_specialization1` for other reasons. One example is this expression:
```jl
Expr(:call, top(:convert), Expr(:call, top(:typeof), SSAValue(28))::DataType, 0)::Any
```
which happens right after building `inference.jl`.

Presumably, I don't want to throw a `MethodError` in such cases, but if we merge `checked_specialization1` into `get_specialization1` then there won't be a way to know why you got `NULL` (and at least the way I've been doing this, that seems key to knowing when the emit the `MethodError`).